an-rope/src/internals.rs

```diff
@@ -55,20 +55,38 @@ impl BranchNode {
     fn split(self, index: usize) -> (Node, Node) {
         let weight = (&self).weight;
         if index < weight {
-            // split the left node
+            // if the index to split on is less than this node's weight,
+            // we split this node's left child.
             let (left, left_right) = self.left.split(index);
+            // the left side of the split left child will become the left side
+            // of the split pair.
+
             let right = if (&left_right).len() == 0 {
+                // if the right side of the split is empty, then the right
+                // side of the returned pair is just this node's right child
                 *self.right
             } else {
+                // otherwise, the right side of the returned pair is a branch
+                // containing the right side of the split node on the left,
+                // and this node's right child on the right
                 Node::new_branch(left_right, *self.right)
             };
             (left, right)
```

Comment:
Calling `split` results in an intermediate state where something_isn't_ in the tree. Returning a `Branch ` would mask that. I think the tuple expresses the intent behind the function much better.